### PR TITLE
CP-15132: Systemd conversion

### PR DESCRIPTION
**Pull request by rafalmiel**

### CODE REVIEW

File: SOURCES/init-xcp-networkd [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#! /bin/bash
```
Comment (robhoes) [inline SOURCES/init-xcp-networkd]: Do we really still need this?

Comment (rafalmiel) [inline SOURCES/init-xcp-networkd]: Same story as xapi, we need for compatibility until testcases are fixed

File: SOURCES/squeezed.service [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Manages host memory by ballooning VMs
+Description=XCP networking daemon
```
Comment (robhoes) [inline SOURCES/squeezed.service]: No...

**Comment by robhoes** [comment]
Of all these, I think that only xcp-networkd needs to be of type notify, and the others can be simple. This is because, for the moment at least, we want to be sure that xcp-networkd sets up networking before xapi starts. All the other ones are decoupled through the message-switch. Xapi needs `After=xcp-networkd.service`, but should not need `After=` for any of the other services in this repo.

File: SOURCES/xcp-networkd.service [hunk 3]
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=XCP networking daemon
+Documentation=man:xcp-networkd(1)
+After=network-online.target forkexecd.service message-switch.service syslog.target
```
Comment (robhoes) [inline SOURCES/xcp-networkd.service]: Please remove `network-online.target` here and below (and anywhere else you see it!).

File: SOURCES/xenopsd-xc.service [hunk 4]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=VM manager for Xen using libxc
+Documentation=man:xenopsd-xc(1)
+After=network-online.target xenstored.service squeezed.service message-switch.service forkexecd.service xapi-storage-script.service syslog.target
```
Comment (robhoes) [inline SOURCES/xenopsd-xc.service]: Remove network-online.target

File: SOURCES/xenopsd-xenlight.service [hunk 5]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=VM manager for Xen using libxenlight
+Documentation=man:xenopsd-xenlight(1)
+After=network-online.target xenstored.service squeezed.service message-switch.service xapi-storage-script.service syslog.target
```
Comment (robhoes) [inline SOURCES/xenopsd-xenlight.service]: network-online.target

File: SPECS/squeezed.spec [hunk 6]
```diff
@@ -23,68 +24,58 @@ BuildRequires:  xen-dom0-libs
...
-#Requires:       redhat-lsb-core
+BuildRequires:  systemd-devel
 Requires:       message-switch
 
-Requires(post): /sbin/chkconfig
-Requires(preun): /sbin/chkconfig
-Requires(preun): /sbin/service
+Requires(post):   systemd
+Requires(preun):  systemd
+Requires(postun): systemd
```
Comment (robhoes) [inline SPECS/squeezed.spec]: %{?systemd_requires}

File: SPECS/squeezed.spec [hunk 7]
```diff
@@ -23,68 +24,58 @@ BuildRequires:  xen-dom0-libs
 BuildRequires:  xen-libs-devel
 BuildRequires:  xen-libs
 BuildRequires:  ocaml-bisect-ppx-devel
-#Requires:       redhat-lsb-core
+BuildRequires:  systemd-devel
```
Comment (robhoes) [inline SPECS/squeezed.spec]: See the other PR. Not needed I think.

File: SPECS/squeezed.spec [hunk 8]
```diff
@@ -23,68 +24,58 @@ BuildRequires:  xen-dom0-libs
...
+- Package for systemd
+
 * Fri Aug 19 2016 Euan Harris <euan.harris@citrix.com> - 0.12.2-1
 - When determining dom0's total memory allocation, always ask Xen first
 
 * Fri Jul 22 2016 Jon Ludlam <jonathan.ludlam@citrix.com> - 0.12.1-1
 - New release
 
-* Mon May 16 2016 Si Beaumont <simon.beaumont@citrix.com> - 0.11.0-2
+* Mon May 16 2016 Si Beaumont <simon.beaumont@citrix.com> - 0.11.0-1
```
Comment (robhoes) [inline SPECS/squeezed.spec]: ?

File: SPECS/xcp-networkd.spec [hunk 9]
```diff
@@ -22,78 +23,64 @@ BuildRequires:  ocaml-xen-api-client-devel
 BuildRequires:  ocaml-netlink-devel
 BuildRequires:  libffi-devel
 BuildRequires:  ocaml-bisect-ppx-devel
+BuildRequires:  ocaml-systemd-devel
+BuildRequires:  systemd-devel
```
Comment (robhoes) [inline SPECS/xcp-networkd.spec]: Do you need this if you have `BuildRequires:  ocaml-systemd-devel`?

File: SPECS/xcp-networkd.spec [hunk 10]
```diff
@@ -22,78 +23,64 @@ BuildRequires:  ocaml-xen-api-client-devel
...
 Requires:       ethtool
 Requires:       libnl3
-#Requires:       redhat-lsb-core
 
-Requires(post): /sbin/chkconfig
-Requires(preun): /sbin/chkconfig
-Requires(preun): /sbin/service
+Requires(post):   systemd
+Requires(preun):  systemd
+Requires(postun): systemd
```
Comment (robhoes) [inline SPECS/xcp-networkd.spec]: use the macro

File: SOURCES/xenopsd-conf [hunk 11]
```diff
@@ -1,43 +0,0 @@
-# Configuration file for xenopsd
```
Comment (robhoes) [inline SOURCES/xenopsd-conf]: Why is this xenopsd.conf file going away? I think we need it!

Comment (robhoes) [inline SOURCES/xenopsd-conf]: Oh I see we have xenopsd-64-conf instead...

File: SPECS/xcp-rrdd.spec [hunk 12]
```diff
@@ -27,57 +28,48 @@ BuildRequires:  xen-libs-devel
 BuildRequires:  blktap-devel
 BuildRequires:  ocaml-bisect-ppx-devel
 #Requires:       redhat-lsb-core
+BuildRequires:  systemd-devel
```
Comment (robhoes) [inline SPECS/xcp-rrdd.spec]: ocaml-systemd?

File: SPECS/xcp-rrdd.spec [hunk 13]
```diff
@@ -27,57 +28,48 @@ BuildRequires:  xen-libs-devel
...
 
-Requires(post): /sbin/chkconfig
-Requires(preun): /sbin/chkconfig
-Requires(preun): /sbin/service
+Requires(post):   systemd
+Requires(preun):  systemd
+Requires(postun): systemd
```
Comment (robhoes) [inline SPECS/xcp-rrdd.spec]: Macro?

File: SPECS/xenopsd.spec [hunk 14]
```diff
@@ -35,14 +33,14 @@ BuildRequires:  xen-dom0-libs-devel
 BuildRequires:  ocaml-uutf-devel
 BuildRequires:  ocaml-xcp-rrd-devel
 BuildRequires:  python-devel
-BuildRequires:  ocaml-bisect-ppx-devel
+BuildRequires:  systemd-devel ocaml-bisect-ppx-devel
```
Comment (robhoes) [inline SPECS/xenopsd.spec]: Hmmm again

File: SPECS/xenopsd.spec [hunk 15]
```diff
@@ -35,14 +33,14 @@ BuildRequires:  xen-dom0-libs-devel
...
 Requires:       message-switch
 Requires:       xenops-cli
 Requires:       xen-dom0-tools
 
-Requires(post): /sbin/chkconfig
-Requires(preun): /sbin/chkconfig
-Requires(preun): /sbin/service
+Requires(post):   systemd
+Requires(preun):  systemd
+Requires(postun): systemd
```
Comment (robhoes) [inline SPECS/xenopsd.spec]: ...

File: SPECS/xenopsd.spec [hunk 16]
```diff
@@ -310,9 +264,6 @@ esac
 - New upstream release that supports coverage analysis
 - Introduce subpackages *-cov for coverage analysis
 
-* Mon May 16 2016 Si Beaumont <simon.beaumont@citrix.com> - 0.12.0-2
```
Comment (robhoes) [inline SPECS/xenopsd.spec]: Still a valid changelog...

**Comment by robhoes** [comment]
Here also, it seems that all `BuildRequires:  systemd-devel` can be replaced by `BuildRequires:  systemd`.

File: SPECS/xcp-networkd.spec [hunk 17]
```diff
@@ -23,15 +23,12 @@ BuildRequires:  ocaml-xen-api-client-devel
 BuildRequires:  ocaml-netlink-devel
 BuildRequires:  libffi-devel
 BuildRequires:  ocaml-bisect-ppx-devel
-BuildRequires:  ocaml-systemd-devel
-BuildRequires:  systemd-devel
+BuildRequires:  ocaml-systemd
```
Comment (robhoes) [inline SPECS/xcp-networkd.spec]: You may still need `BuildRequires:  systemd`... let's see if it works on your branch...

**Pull request merged**